cubeviz/controls/wavelengths.py

```diff
@@ -88,7 +84,9 @@ def update_units(self, units):
         self._send_wavelength_unit_message(units)
         self._send_wavelength_message(self._wavelengths)
 
-        specviz_dispatch.changed_units.emit(x=units)
+        self._cv_layout.specviz._widget.hub.plot_widget.spectral_axis_unit = units
+        self._cv_layout.specviz._widget._slice_indicator.setPos(
+            self._wavelengths[self._cv_layout._active_cube._widget.slice_index])
```

Comment:
Just curious but why are these lines necessary? I would have thought that this would have been handled by the two messages that were sent above.

Reply:
There's a race condition with the messages sent out from cubeviz due to the bi-directional connection on the slice index indicators. I'll push a workaround, but it's not _great_ (uses signal blockers), but slightly more in line with what you're getting at here.